### ISSUE TITLE
[A11ySev2] [Screen Reader -Dashboards with Grafana -Add another Transformations]: Screen reader does not announce the pressed state for buttons under the "Add Another Transformation" tab.

### ISSUE THREAD

**Issue by LinglingTong** [grafana/grafana#118758]
## Steps to reproduce

Repro Steps: 
  
 
1. Open 'Azure Monitor dashboards with Grafana' in Azure Portal
2. "Monitor | Dashboards with Grafana | Gallery" screen appears with already Dashboards with Grafana tab selected.
3. Navigate to the all the controls available on Dashboards with Grafana blade like tool bar , controls under All Dashboards tab.
4. Navigate New menu present in the tool bar and select New Dashboard.
5. Navigate to "Add visualization" button and hit ENTER key. "Select data source" popup will appear.
6. Navigate to the TestData list item select it. 
7. Navigate to the Transformation tab and select add transformation button.
8. Navigate to all controls under Add another transformation dialog.
9. Verify the issue whether screen reader is announcing the state for buttons under Add another Transformation tab or not.

**Actual Result:**
The screen reader does not announce whether the buttons are pressed or unpressed, leaving users unable to understand their current state. 
**Observation**: 
The buttons under the "Add Another Transformation" tab do not provide the aria-pressed attribute or equivalent functionality. As a result, screen readers fail to announce the pressed state of these buttons, making them inaccessible to users with visual impairments. 

**Expected Result:**
The buttons should have an accessible state that is announced by screen readers. When a button is pressed, the screen reader should announce "pressed," and when it is unpressed, it should announce "not pressed." This can be achieved by implementing the aria-pressed attribute on the buttons.

**Relevant WCAG Criteria:** [4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html) (Level A)

## Environment

- **Grafana version:** 12.2.0
- **Data source type & version:**
- **User OS & Browser:** Version 25H2 (OS Build 26200.7093) / Chrome Version 142.0.7444.177 (Official Build) (64-bit)
- **Others:**

**Comment by LinglingTong** [comment]
<img width="1919" height="1071" alt="Image" src="https://github.com/user-attachments/assets/8510d0b1-540a-4256-ae89-1c0596c4d79a" />
<img width="1919" height="1106" alt="Image" src="https://github.com/user-attachments/assets/64df46e3-f5ad-4436-ba7a-bdda5050ed5f" />

**Comment by v-kavyagupta** [comment]
We verified this issue on the URL:http://localhost:3000/dashboard/new?orgId=1&from=now-7d&to=now&timezone=browser&var-adhoc0=&dtab=new-tab&editPanel=2&tab=transformations
Issue dies not repro, Screen reader does not announce the pressed state for buttons under the "Add Another Transformation" tab.

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/4b671945-63d4-4fda-92e0-3b8439145d07" />
